src/Data/NonEmpty.purs

```diff
@@ -167,3 +167,11 @@ instance foldable1NonEmpty :: Foldable f => Foldable1 (NonEmpty f) where
 
 instance unfoldable1NonEmpty :: Unfoldable f => Unfoldable1 (NonEmpty f) where
   unfoldr1 f b = uncurry (:|) $ unfoldr (map f) <$> f b
+
+-- | For `Semigroup` we don't know for sure that `pure` will behave sensibly 
+-- | alongside `<>` because we don't have any laws which govern their
+-- | behaviour, in practice, with Array, List, it should behave sensibly.
```

Comment:
What do you think of a slight rephrase that asserts that it's OK for you to use this with arrays, lists, and so on, but you may need to be careful with other types?

```suggestion
-- | This is a lawful `Semigroup` instance that will behave sensibly for common nonempty
-- | containers like lists and arrays. However, it's not guaranteed that `pure` will behave
-- | sensibly alongside `<>` for all types, as we don't have any laws which govern their behavior.
```